examples/run/sycl/seq_example_sycl.sycl

```diff
@@ -434,29 +450,27 @@ int seq_run(const traccc::opts::detector& detector_opts,
         sd_performance_writer.finalize();
     }
 
-    std::cout << "==> Statistics ... " << std::endl;
-    std::cout << "- read    " << n_cells << " cells" << std::endl;
-    std::cout << "- created (cpu)  " << n_measurements << " measurements"
-              << std::endl;
-    std::cout << "- created (cpu)  " << n_spacepoints << " spacepoints"
-              << std::endl;
-    std::cout << "- created (sycl) " << n_spacepoints_sycl
-              << " spacepoints     " << std::endl;
-
-    std::cout << "- created  (cpu) " << n_seeds << " seeds" << std::endl;
-    std::cout << "- created (sycl) " << n_seeds_sycl << " seeds" << std::endl;
-    std::cout << "- found (cpu)    " << n_found_tracks << " tracks"
-              << std::endl;
-    std::cout << "- found (sycl)   " << n_found_tracks_sycl << " tracks"
-              << std::endl;
-    std::cout << "==>Elapsed times...\n" << elapsedTimes << std::endl;
+    TRACCC_INFO("==> Statistics ... ");
+    TRACCC_INFO("- read    " << n_cells << " cells");
+    TRACCC_INFO("- created (cpu)  " << n_measurements << " measurements");
+    TRACCC_INFO("- created (cpu)  " << n_spacepoints << " spacepoints");
+    TRACCC_INFO("- created (sycl) " << n_spacepoints_sycl
+                                    << " spacepoints     ");
+
+    TRACCC_INFO("- created  (cpu) " << n_seeds << " seeds");
+    TRACCC_INFO("- created (sycl) " << n_seeds_sycl << " seeds");
+    TRACCC_INFO("- found (cpu)    " << n_found_tracks << " tracks");
+    TRACCC_INFO("- found (sycl)   " << n_found_tracks_sycl << " tracks");
+    TRACCC_INFO("==>Elapsed times...\n" << elapsedTimes);
 
     return 0;
 }
 //
 // The main routine
 //
 int main(int argc, char* argv[]) {
+    std::unique_ptr<const traccc::Logger> logger = traccc::getDefaultLogger(
+        "TracccExampleSeqSycl", traccc::Logging::Level::INFO);
```

Comment:
I might very well not understand the Acts logging code...

But providing default logger objects to the algorithms with this formalism would not be possible? What sort of a global state would that mean, if by default they all get their own loggers printing to `std::cout` at an `INFO` level?

The user would still be able to override everything at their pleasure. But it would seem possible to provide meaningful default objects to the algorithms. :thinking: What am I missing here?

Reply:
Right, but we cannot print to users' `stdout` without their permission; that's just bad software design. If we want to provide a default logger (which I agree we should) then the only proper way to do that is to provide a default logger that doesn't print anything.

The only remotely viable alternative is to to have a traccc-global logger which the user can set which is then used for all instances where a traccc function needs a default. But as I said I think we should aim to avoid this kind of setup.